File: cesar.js
function cifradoCesar(texto, desplazamiento) {
  let resultado = "";

  for (let i = 0; i < texto.length; i++) {
    let char = texto[i];

    if (/[a-zA-Z]/.test(char)) {//test comprueba si el caracter es una letra
      let base = char === char.toUpperCase() ? 65 : 97;//diferenciamos entre mayusculas y minusculas
      let codigo = texto.charCodeAt(i);//obtenemos el codigo ASCII del caracter

      resultado += String.fromCharCode(((codigo - base + desplazamiento) % 26 + 26) % 26 + base);
    } else {
      resultado += char; // Deja espacios y símbolos sin cambiar
    }
  }
  return resultado;
}

// Capturar mensaje desde consola
process.stdin.on("data", function(data) {
  let mensaje = data.toString().trim();
  let clave = 3;

  let cifrado = cifradoCesar(mensaje, clave);
  let descifrado = cifradoCesar(cifrado, -clave);

  console.log("Mensaje original:", mensaje);
  console.log("Cifrado:", cifrado);
  console.log("Descifrado:", descifrado);

  process.exit(); // <- cerramos la app después de imprimir
});

File: cesar.py
def cifrado_cesar(texto, desplazamiento):
    resultado = ""

    for char in texto:
        if char.isalpha():  # letras
            base = ord('A') if char.isupper() else ord('a')
            resultado += chr((ord(char) - base + desplazamiento) % 26 + base)

        elif char.isdigit():  # números
            base = ord('0')
            resultado += chr((ord(char) - base + desplazamiento) % 10 + base)

        else:  # espacios, símbolos, etc.
            resultado += char

    return resultado


# Ejemplo de uso
if __name__ == "__main__":
    mensaje = input("Introduce el mensaje a cifrar: ")
    clave = 3 # Desplazamiento fijo de 3 para el cifrado César

    cifrado = cifrado_cesar(mensaje, clave)
    descifrado = cifrado_cesar(cifrado, -clave)

    print("Mensaje original:", mensaje)
    print("Cifrado:", cifrado)
    print("Descifrado:", descifrado)
